### ISSUE TITLE
CPU Usage > 100% constantly

### ISSUE THREAD

**Issue by markmandel** [NicolasConstant/sengi#522]
Sengi version: 1.2.0

```
❯ lsb_release --all
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux bookworm/sid
Release:        n/a
Codename:       bookworm
```

I run 6 columns:

1. home
2. local timeline (account 1)
3. local timeline (account 2)
4. local timeline (account 3)
5. Hashtag query
6. Hashtag query

Sengi usually takes up around 150-350% cpu on my 8 core machine.

Visualising with `btop`:

![image](https://user-images.githubusercontent.com/298370/207705044-ef2c1700-2541-4f0b-8216-8e5837fd8ffa.png)

While Running Sengi:

![image](https://user-images.githubusercontent.com/298370/207705152-ef228155-a06c-4e3c-aa33-de03ee1ccf58.png)

Then, after I exit it:
![image](https://user-images.githubusercontent.com/298370/207705365-d8c824b5-7053-4c73-a43f-f3f050bebe8d.png)

Not sure what I can do to profile, but happy to do so if it helps.

Some ideas:
* Gif playback. Wondering if they all playback all the time? 
* Something to do with rendering, but I've no idea what.

Still love Sengi, but struggling to keep it open all the time with this CPU usage.

**Comment by NicolasConstant** [comment]
Weird. What build are you using?

**Comment by markmandel** [comment]
Sengi version: 1.2.0, installed from deb.

One thing I did notice when I installed the deb, was that it _seemed like_  it was installing the 1.1.4 version:

```shell
❯ sudo gdebi  Sengi-1.2.0-linux.deb
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Reading state information... Done

A multi-account desktop client for Mastodon and Pleroma
Do you want to install the software package? [y/N]:Y
/usr/bin/gdebi:113: FutureWarning: Possible nested set at position 1
  c = findall("[[(](\S+)/\S+[])]", msg)[0].lower()
(Reading database ... 560903 files and directories currently installed.)
Preparing to unpack Sengi-1.2.0-linux.deb ...
Unpacking sengi (1.1.4-1087) over (1.1.4-1087) ...
Setting up sengi (1.1.4-1087) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for mailcap (3.70+nmu1) ...
Processing triggers for gnome-menus (3.36.0-1) ...
Processing triggers for desktop-file-utils (0.26-1) ...
```

**Comment by NicolasConstant** [comment]
Yeah the electron build can be another version but it has no importance: the version of Sengi is the one loaded in the UI/Settings.

Can you try the webapp version to check if it's an Electron issue or something linked to a script going crazy?

**Comment by markmandel** [comment]
That gave me a good idea - here is a 1 minute profile of the Electron App, to see if you can see anything that stands out.
[profile.zip](https://github.com/NicolasConstant/sengi/files/10232542/profile.zip)

**Comment by markmandel** [comment]
So I just installed Sengi as a PWA - and wow, so much less CPU usage!

![image](https://user-images.githubusercontent.com/298370/207949469-5e5aeb83-1728-4600-9dad-87bb7e5e0923.png)

![image](https://user-images.githubusercontent.com/298370/207949551-e5e900c5-1359-4905-a9aa-4a684b23dbbd.png)

![image](https://user-images.githubusercontent.com/298370/207949598-aac2b5ca-4da3-4350-8e7b-207c5fcd2bc3.png)

So clearly that's how I'll be using it form now on 😄 

But I also found something else out that I think is a big culprit.

(Going back to the Electron App)

My fourth column is a search for #gamedev - and if I remove that, CPU usage drops to 10-30% (totally acceptable levels)

I don't know if it's just because that has way more gifs and videos, or it's a popular tag, or what - but it makes a huge difference if it's there or not.

![image](https://user-images.githubusercontent.com/298370/207951918-3f5a193c-7b6a-49bd-a448-4ae9d68ada38.png)

**Comment by markmandel** [comment]
The plot thicks further! I left it running for a while, and the PWA started showing way more CPU usage (similar to the electron app)!

![image](https://user-images.githubusercontent.com/298370/207974317-e9c94da8-d062-441c-aaa7-ff17f5ce1bc0.png)

So it definitely gets worse over time.

I dropped the #gamedev column, and saw a big CPU drop (although still ~90% or so). I'm thinking it's something to do with hashtag columns, and maybe ones that tend to be gif/video heavy.

![image](https://user-images.githubusercontent.com/298370/207974613-b52bdcf2-d564-47a0-a088-734b179f79ed.png)

If I reload the PWA page, then CPU also drops back down - so it does seem incremental over time.

![image](https://user-images.githubusercontent.com/298370/207974906-59e65039-0077-458e-a80d-f7cd95ad15fc.png)

**Comment by rpetti** [comment]
I think each column will keep at least 40 posts loaded at all times, and I don't believe it will pause playback when posts aren't visible. Could that account for the large CPU usage? I also wonder if GPU acceleration (or the lack thereof) is a factor here.

**Comment by markmandel** [comment]
> I think each column will keep at least 40 posts loaded at all times, and I don't believe it will pause playback when posts aren't visible. Could that account for the large CPU usage? 

That was my thought. I had noticed Tweetdeck does pause gifs/videos (or give you the option to only play on mouseover).

This machine is running an Nvidia card... so it should be GPU accelerated.
